src/server/project.js

```diff
@@ -0,0 +1,291 @@
+            } else {
+                this.pluginsTelemetry.simulatedNonBuiltIn.push(pluginId);
+            }
+        } else {
+            this.pluginsTelemetry.notSimulated.push(pluginId);
+        }
+    }.bind(this));
+
+    // Find built-in debug-host plugins (plugins that should be part of the simulation due to the presence of a
+    // debug-host, even if they weren't added to the project).
+    var simulatedDebugHostPlugins = {};
+
+    if (this.debugHostHandlers) {
+        var rawBuiltInPluginList = utils.getDirectoriesInPath(dirs.plugins);
+
+        rawBuiltInPluginList.forEach(function (pluginId) {
+            if (simulatedDefaultPlugins[pluginId] || simulatedProjectPlugins[pluginId]) {
+                return;
+            }
+
+            var pluginPath = path.join(dirs.plugins, pluginId);
+
+            if (pluginPath && pluginUtil.shouldUsePluginWithDebugHost(pluginPath, null, this.debugHostHandlers)) {
+                simulatedDebugHostPlugins[pluginId] = pluginPath;
+                this.pluginsTelemetry.simulatedBuiltIn.push(pluginId);
+            }
+        }.bind(this));
+    }
+
+    // Register the plugins. The default plugins are first, then the debug-host plugins (to have the correct order in
+    // sim-host).
+    var that = this;
+    function registerPlugins(pluginDictionary) {
+        Object.keys(pluginDictionary).forEach(function (pluginId) {
+            that.plugins[pluginId] = pluginDictionary[pluginId];
+        });
+    }
+
+    registerPlugins(simulatedDefaultPlugins);
+    registerPlugins(simulatedDebugHostPlugins);
+    registerPlugins(simulatedProjectPlugins);
+
+    this._simulator.telemetry.sendTelemetry( 'plugin-list', {
+        simulatedBuiltIn: this.pluginsTelemetry.simulatedBuiltIn
+    }, {
+        simulatedNonBuiltIn: this.pluginsTelemetry.simulatedNonBuiltIn,
+        notSimulated: this.pluginsTelemetry.notSimulated
+    });
+    this._addPlatformDefaultHandlers();
+    this._populateRouter();
+};
+
+Project.prototype.getPlugins = function () {
+    return this.plugins;
+};
+
+Project.prototype.getRouter = function () {
+    this._router = this._router || cordovaServe.Router();
+    return this._router;
+};
+
+Project.prototype.getPluginsTelemetry = function () {
+    return this.pluginsTelemetry;
+};
+
+/**
+ * Prepares the project and initializes the simulation plugin list.
+ *
+ * @param {Object=} currentState (Optional) The current state of the project for caching purposes.
+ */
+Project.prototype.prepare = function() {
+    if (!this._preparePromise) {
+        var d = Q.defer();
+        var currentProjectState;
+
+        this._preparePromise = d.promise;
+        this._lastPlatform = this.platform;
+
+        this._getProjectState().then(function (currentState) {
+            currentProjectState = currentState || this._getProjectState();
+            var previousState = this._previousPrepareStates[this.platform];
```

Comment:
This variable could be refactored out of the promise chain so that it can be reused further down instead of defining it again. This is not a huge deal though, just something that crossed my mind :)